### PR TITLE
remove buggy shell script in favor of well-tested JS lib

### PR DESCRIPTION
**Pull request by randallsquared**
The wait-for.sh script previously included with nb-ms

- claims to take `-t 0` to wait forever, but in fact waits 2 seconds in that case,
- apparently destroys environment variables

These are merely symptoms, however; the root cause of these issues is that this shell script is far from the domain of expertise of most node developers, not automatically tested, and not packaged for distribution of updates.   All of these but the first is solvable by more shell code, but in the meantime we have a different solution to all of them: just use JS.

The `waitstaff` lib handles the same use case with a superset of the claimed arguments, and is a well-tested library with high coverage.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -22,11 +22,11 @@ services:
...
       - NODE_LOGGER_GRANULARLEVELS=1
       - NODE_CONFIG_DISABLE_FILE_WATCH="Y"
-      #- PORT=5501 # this is overriden in shell/start-dev.sh since wait-for.sh compromises it from here
+      - PORT=5501
     # command: nodemon -e js,coffee,jade,handlebars server.js
     # @see: https://docs.docker.com/compose/startup-order/
-    # command: ["./shell/wait-for.sh", "ms-nodebootstrap-example-db:3306", "--", "./shell/start-dev.sh"]
-    command: ./shell/wait-for.sh ms-nodebootstrap-example-db:3306 -- ./shell/start-dev.sh
+    # wait up to 15 seconds for mysql to become available
+    command: npx waitstaff -t=15000 ms-nodebootstrap-example-db:3306 -- ./shell/start-dev.sh
```
Comment (inadarei) [inline docker-compose.yml]: Very minor, but - given how package.json does contain waitstaff explicitly, why `npx` and not just `npm`? The only difference, in this case, is that npx would install waitstaff if not found, where npm - would not, right?

**Comment by inadarei** [comment]
To be fair - wait-for.sh is straight from docker.com documentation that describes what should be done in case a container needs to wait for another container :) So it is within the domain expertise of Docker users and doesn't need to be in the domain expertise of Node devs (since it is nothing Node-specific).

That said, I won't argue that wait-sh can be buggy. However, why did you decide to go with something that you just wrote and, on the basis of being very new can also be buggy, instead of going what seems to be a fairly widely used, already existing module: https://www.npmjs.com/package/wait-on

**Comment by randallsquared** [comment]
`wait-on` would have been a good choice, for sure: it has nearly every feature I would want to add to `waitstaff`, and many, many more besides.   I wanted something that added less bulk to node_modules (72K vs 20M), but the importance of that is arguable, for sure (especially since it's only used in dev in nb-ms).   

When I started working on `waitstaff` a few weeks ago, I was just replacing `wait-for.sh`, which I see slightly different versions of all over the place, and which has a different next-command interface than `wait-on`.  It grew to be slightly more like `wait-on` as I wrote it, but is still significantly cheaper, for whatever that is worth.

File: docker-compose.yml (hunk 1 above)
Comment (inadarei) [reply docker-compose.yml]: Thanks. Makes sense

**Comment by randallsquared** [comment]
I'd think replacing a shell script with a Go solution in an otherwise JS codebase would feel even more wrong.  :)   

I'm going to close the PR.   Let's discuss a better solution in the future.

**Comment by inadarei** [comment]
Um, I am afraid that is the fundamental difference between our views on this scrip. I don't think it's part of the Node code, rather it is part of the Docker setup (and should be similar if this were a Go project needing to wait for a db port, or a python project). Once and if you see it in that way - using Go to implement it makes more sense. Go is replacing shell scripts everywhere, especially in container-space, no? :)

**Pull request closed without merging**